apps/api/src/scraper/scrapeURL/lib/extractMetadata.ts

```diff
@@ -0,0 +1,132 @@
+  let dcDescription: string | undefined = undefined;
+  let dcTermsKeywords: string | undefined = undefined;
+  let modifiedTime: string | undefined = undefined;
+  let publishedTime: string | undefined = undefined;
+  let articleTag: string | undefined = undefined;
+  let articleSection: string | undefined = undefined;
+  const customMetadata: Record<string, string | string[]> = {};
+
+  const soup = load(html);
+
+  try {
+    title = soup("title").text() || undefined;
+    description = soup('meta[name="description"]').attr("content") || undefined;
+    
+    // Assuming the language is part of the URL as per the regex pattern
+    language = soup('html').attr('lang') || undefined;
+
+    keywords = soup('meta[name="keywords"]').attr("content") || undefined;
+    robots = soup('meta[name="robots"]').attr("content") || undefined;
+    ogTitle = soup('meta[property="og:title"]').attr("content") || undefined;
+    ogDescription = soup('meta[property="og:description"]').attr("content") || undefined;
+    ogUrl = soup('meta[property="og:url"]').attr("content") || undefined;
+    ogImage = soup('meta[property="og:image"]').attr("content") || undefined;
+    ogAudio = soup('meta[property="og:audio"]').attr("content") || undefined;
+    ogDeterminer = soup('meta[property="og:determiner"]').attr("content") || undefined;
+    ogLocale = soup('meta[property="og:locale"]').attr("content") || undefined;
+    ogLocaleAlternate = soup('meta[property="og:locale:alternate"]').map((i, el) => soup(el).attr("content")).get() || undefined;
+    ogSiteName = soup('meta[property="og:site_name"]').attr("content") || undefined;
+    ogVideo = soup('meta[property="og:video"]').attr("content") || undefined;
+    articleSection = soup('meta[name="article:section"]').attr("content") || undefined;
+    articleTag = soup('meta[name="article:tag"]').attr("content") || undefined;
+    publishedTime = soup('meta[property="article:published_time"]').attr("content") || undefined;
+    modifiedTime = soup('meta[property="article:modified_time"]').attr("content") || undefined;
+    dcTermsKeywords = soup('meta[name="dcterms.keywords"]').attr("content") || undefined;
+    dcDescription = soup('meta[name="dc.description"]').attr("content") || undefined;
+    dcSubject = soup('meta[name="dc.subject"]').attr("content") || undefined;
+    dcTermsSubject = soup('meta[name="dcterms.subject"]').attr("content") || undefined;
+    dcTermsAudience = soup('meta[name="dcterms.audience"]').attr("content") || undefined;
+    dcType = soup('meta[name="dc.type"]').attr("content") || undefined;
+    dcTermsType = soup('meta[name="dcterms.type"]').attr("content") || undefined;
+    dcDate = soup('meta[name="dc.date"]').attr("content") || undefined;
+    dcDateCreated = soup('meta[name="dc.date.created"]').attr("content") || undefined;
+    dcTermsCreated = soup('meta[name="dcterms.created"]').attr("content") || undefined;
+
+    try {
+      // Extract all meta tags for custom metadata
```

Comment:
why would this not work for the base cases?

Reply:
What do you mean?

Reply:
Wont the way you extract the custom ones work for the predefined ones above too?

Reply:
It does not retain compatibility for WebScraper, and breaks tests. This was also in WebScraper.